easyaudit/models.py

```diff
@@ -18,82 +18,123 @@ class CRUDEvent(models.Model):
+        (DELETE, _("Delete")),
+        (M2M_CHANGE, _("Many-to-Many Change")),
+        (M2M_CHANGE_REV, _("Reverse Many-to-Many Change")),
+        (M2M_ADD, _("Many-to-Many Add")),
+        (M2M_ADD_REV, _("Reverse Many-to-Many Add")),
+        (M2M_REMOVE, _("Many-to-Many Remove")),
+        (M2M_REMOVE_REV, _("Reverse Many-to-Many Remove")),
+        (M2M_CLEAR, _("Many-to-Many Clear")),
+        (M2M_CLEAR_REV, _("Reverse Many-to-Many Clear")),
     )
 
-    event_type = models.SmallIntegerField(choices=TYPES, verbose_name=_('Event type'))
-    object_id = models.CharField(max_length=255, verbose_name=_('Object ID'))
-    content_type = models.ForeignKey(ContentType, on_delete=models.CASCADE, db_constraint=False, verbose_name=_('Content type'))
-    object_repr = models.TextField(null=True, blank=True, verbose_name=_('Object representation'))
-    object_json_repr = models.TextField(null=True, blank=True, verbose_name=_('Object JSON representation'))
-    changed_fields = models.TextField(null=True, blank=True, verbose_name=_('Changed fields'))
-    user = models.ForeignKey(settings.AUTH_USER_MODEL, null=True,
-                             blank=True, on_delete=models.SET_NULL,
-                             db_constraint=False, verbose_name=_('User'))
-    user_pk_as_string = models.CharField(max_length=255, null=True, blank=True,
-                                         help_text=_('String version of the user pk'), verbose_name=_('User PK as string'))
-    datetime = models.DateTimeField(auto_now_add=True, verbose_name=_('Date time'))
+    event_type = models.SmallIntegerField(choices=TYPES, verbose_name=_("Event type"))
+    object_id = models.CharField(max_length=255, verbose_name=_("Object ID"))
+    content_type = models.ForeignKey(
+        ContentType,
+        on_delete=models.CASCADE,
+        db_constraint=False,
+        verbose_name=_("Content type"),
+    )
+    object_repr = models.TextField(
+        default="", blank=True, verbose_name=_("Object representation")
+    )
+    object_json_repr = models.TextField(
+        default="", blank=True, verbose_name=_("Object JSON representation")
+    )
+    changed_fields = models.TextField(
+        default="", blank=True, verbose_name=_("Changed fields")
+    )
+    user = models.ForeignKey(
+        settings.AUTH_USER_MODEL,
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+        db_constraint=False,
+        verbose_name=_("User"),
+    )
+    user_pk_as_string = models.CharField(
+        max_length=255,
+        default="",
+        blank=True,
+        help_text=_("String version of the user pk"),
+        verbose_name=_("User PK as string"),
+    )
+    datetime = models.DateTimeField(auto_now_add=True, verbose_name=_("Date time"))
+
+    class Meta:
+        verbose_name = _("CRUD event")
+        verbose_name_plural = _("CRUD events")
+        ordering = ["-datetime"]
+        indexes = [models.Index(fields=["object_id", "content_type"])]
```

Comment:
I don't think it's a necessary change, and I agree that #258 should likely merge first, since this is duplicated work. I'll just fix the merge conflicts after those two merge.